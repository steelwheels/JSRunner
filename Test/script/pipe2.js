/**
 * pipe1.js
 */

let pipe0 = Pipe() ;
let pipe1 = Pipe() ;

console.log("[allocate process]\n") ;
let process = system("/bin/cat", pipe0.reading, pipe1.writing, stderr) ;

if(process == null){
	console.log("[Error] Could not launch command\n") ;
	exit(1) ;
}

/* send input */
console.log("[send input]\n") ;
pipe0.writing.put("Input from JavaScript !!\n") ;
pipe0.writing.close() ;

/* receive output */
console.log("[receive output]\n") ;
let c = pipe1.reading.getc() ;
while(c != null){
	console.log(`[receive] ${c}\n`) ;
	c = pipe1.reading.getc() ;
}

process.waitUntilExit() ;

console.log("[bye]\n");

