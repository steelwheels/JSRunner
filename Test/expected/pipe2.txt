[allocate process]
[send input]
[receive output]
[receive] I
[receive] n
[receive] p
[receive] u
[receive] t
[receive]  
[receive] f
[receive] r
[receive] o
[receive] m
[receive]  
[receive] J
[receive] a
[receive] v
[receive] a
[receive] S
[receive] c
[receive] r
[receive] i
[receive] p
[receive] t
[receive]  
[receive] !
[receive] !
[receive] 

[bye]
